src/main/java/rainy/tasks/TaskTracker.java

```diff
@@ -0,0 +1,221 @@
+package rainy.tasks;
+
+import java.util.ArrayList;
+import java.util.Collections;
+
+import rainy.database.UI;
+import rainy.rainyexceptions.InvalidIndexException;
+import rainy.rainyexceptions.InvalidMarkAndUnmarkException;
+
+/**
+ * Keeps track of the list of classes and variables pertaining to how many tasks there are.
+ *
+ */
+public class TaskTracker {
+    private ArrayList<Task> taskList;
```

Comment:
Perhaps you could rename this to `tasks` to since this is a collection of Tasks and it should be plural?